beacon-chain/cache/depositsnapshot/deposit_tree_snapshot.go

```diff
@@ -60,3 +50,36 @@ func fromTreeParts(finalised [][32]byte, depositCount uint64, executionBlock exe
 	snapshot.depositRoot = root
 	return snapshot, nil
 }
+
+// ToProto converts the underlying trie into its corresponding proto object.
+func (ds *DepositTreeSnapshot) ToProto() *protodb.DepositSnapshot {
+	tree := &protodb.DepositSnapshot{
+		Finalized:      make([][]byte, len(ds.finalized)),
+		DepositRoot:    ds.depositRoot[:],
```

Comment:
there is a safecopy bytes below, is this something you want to do here?

Reply:
Thanks, done